stl/inc/memory

```diff
@@ -456,9 +453,15 @@ namespace ranges {
                 return _First;
             }
 
-            auto _UFirst = _Get_unwrapped_n(_STD move(_First));
-            for (; _Count-- > 0; ++_UFirst) {
-                _RANGES destroy_at(_STD addressof(*_UFirst));
+            auto _UFirst = _Get_unwrapped_n(_STD move(_First), _Count);
+            if constexpr (is_trivially_destructible_v<iter_value_t<_It>>) {
+                _RANGES advance(_UFirst, _Count);
+            } else {
+                do {

```

Comment:
No change requested. I hate do while loops. 

It is even 2 lines longer. Think about all that screen real estate

Reply:
Indeed, `do-while` loops aren't our preferred form - but we'll use them when they can potentially save cycles. They're a signal that we've already established the loop invariant at the beginning.

Reply:
```c++
for (...) {
}
```
2 lines.
```c++
while (...) {
}
```
2 lines.
```c++
do {
} while(...);
```
2 lines.

Fun aside, like Stephan says we use do/while when the invariant has already been validated elsewhere like this,